lib/utils/watchers.js

```diff
@@ -109,23 +110,38 @@ export function newSitetable(siteTable: HTMLElement) {
 	return registerElement('siteTable', siteTable, sorter);
 }
 
-function addCommentsObserver(ele) {
-	const observer = observe(ele, { childList: true }, mutation => {
+async function addCommentsObserver(ele) {
+	let sitetable: ?HTMLElement = Array.from(ele.children).find(e => e.classList.contains('sitetable'));
+
+	// Comments without replies does not have `.sitetable`
+	if (!sitetable) {
+		await new Promise(resolve => {
+			const observer = observe(ele, { childList: true }, mutation => {
+				if (!mutation.addedNodes.length) return;
+				const addedNode = downcast(mutation.addedNodes[0], HTMLElement);
+				if (!addedNode.classList.contains('sitetable')) return;
+
+				sitetable = addedNode;
+				observer.disconnect();
+				resolve();
+
+				for (const e of addedNode.querySelectorAll('.thing')) {
+					registerElement('newComments', e);
+				}
+			});
+		});
+	}
+
+	/*:: if (!sitetable) throw new Error(); */
```

Comment:
You might be able to convince Flow if you resolve the promise with addedNode and assign it outside

```js
if (!sitetable) {
		sitetable = await new Promise(resolve => {
```